server/api/lib/tags/index.js

```diff
@@ -0,0 +1,62 @@
+ const { tags, tagDataRules } = require('../../constants');
+ const PlanAreaChanges = require('../../../api/model/plan_area_changes');
+
+ const  generateTagsForPlan = async (planId) => {
+    let planTags = [];
```

Comment:
please change this let to const

Reply:
but I add things to it, so it's not a const...

Reply:
Cost and let are referring to the reference. The reference is set once at your code, on deceleration time.
After that you are mutating the object, but don't change the reference. In that scenario, const is preferable. 

Legit code using const:
```
const a = [];
a.push(2);  //mutating the object
```
Non legit code that will fail:
```
const a = [];
a = [2];   // mutate the reference
```